### DISCUSSION TITLE
Separate updates for the same dependency in monorepo (with ArgoCD configuration)

### DISCUSSION THREAD

**Discussion by szpak** [renovatebot/renovate#22434] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.40.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

There is a monorepo with ArgoCD configuration for different project withing a group. Some projects use the same Helm chart and Renovate detects that and propose one update in all the places. In general, it is nice, but for monorepo, one could deploy new chart version on one environment, test it and rollout it further. Therefore, it would be more convenient here to have separate MRs for all the changes (even though it is the same dependency in different files). Is it possible to achieve?

```
foobar-env-config
- dev
- -app1-dev.yaml
- -app2-dev.yaml
- prod
- -app1-prod.yaml
- -app2-prod.yaml
```

Every YAML file refers to:
```
  source:
    path: .
    repoURL: git@git.example.com:devops/charts/my-chart.git
    targetRevision: v1.2.0
    helm:
        ...
```

my-chart version in every file should be updated in separate MRs.

As a bonus, it would be good to be able to group somehow the files to - for example - update the version in all files in a "/devel" directory in one MR and in the second in all files in "/production" directory. Nevertheless, MR per dependency would be enough for that case.

### What you tried so far.

I've checked in the documentation for `matchPaths` and grouping, but maybe grouping per one dependency could be disabled at all?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
Search this repo for discussions about `additionalBranchPrefix` and `parentDir` and see if that's what you need

**Reply by szpak** [reply]
Yes, that is something what I needed, thanks! In addition to `parentDir` also `packageFile` can be handy to force a PR per project, if needed.

For a reference for others bumping into that case:
```
  packageRules: [
    {
      // "matchDatasources" matcher could be strengthened, if needed
      matchDatasources: ["git-tags"],
      // or "{{packageFile}}-" or "{{parentDir}}-{{packageFile}}" to ungroup updates even more, if needed
      additionalBranchPrefix: "{{parentDir}}-",
      commitMessagePrefix: "[{{parentDir}}] "
    }
  ]
```

**Reply by ruben-chainalysis** [reply]
This is the second time I stumble into this discussion, hopefully it's useful for others if not just future me.
If your directory structure has multiple environments in the same repo (not sure if this is common in ArgoCD, but it is in FluxCD), just `parentDir` or a combination of it with `packageFile` might leave you with a single PR for all environments. To get the full path, use `packageFileDir` and you should get separate PRs.
